src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/contactperson/ContactPersonMappingApiMockServer.kt

```diff
@@ -325,6 +334,113 @@ class ContactPersonMappingApiMockServer(private val objectMapper: ObjectMapper)
     )
   }
 
+  fun stubGetByDpsContactPhoneIdOrNull(
+    dpsContactPhoneId: Long = 123456,
+    mapping: PersonPhoneMappingDto? = PersonPhoneMappingDto(
+      nomisId = 654321,
+      dpsId = dpsContactPhoneId.toString(),
+      dpsPhoneType = PersonPhoneMappingDto.DpsPhoneType.PERSON,
+      mappingType = PersonPhoneMappingDto.MappingType.MIGRATED,
+    ),
+  ) {
+    mapping?.apply {
+      mappingServer.stubFor(
+        get(urlEqualTo("/mapping/contact-person/phone/dps-contact-phone-id/$dpsContactPhoneId")).willReturn(
+          aResponse()
+            .withHeader("Content-Type", "application/json")
+            .withStatus(HttpStatus.OK.value())
+            .withBody(objectMapper.writeValueAsString(mapping)),
+        ),
+      )
+    } ?: run {
+      mappingServer.stubFor(
+        get(urlEqualTo("/mapping/contact-person/phone/dps-contact-phone-id/$dpsContactPhoneId")).willReturn(
+          aResponse()
+            .withHeader("Content-Type", "application/json")
+            .withStatus(HttpStatus.NOT_FOUND.value())
+            .withBody(objectMapper.writeValueAsString(ErrorResponse(status = 404))),
+        ),
+      )
+    }
+  }
+
+  fun stubGetByDpsContactAddressPhoneIdOrNull(
+    dpsContactAddressPhoneId: Long = 123456,
+    mapping: PersonPhoneMappingDto? = PersonPhoneMappingDto(
+      nomisId = 654321,
+      dpsId = dpsContactAddressPhoneId.toString(),
```

Comment:
Strange to put a number in a string?

Reply:
it s because the mapping always has Strings for Ids but the real service currently returns a number of(rare, given most services use UUID)